### ISSUE TITLE
Displayed encumbrance effects don't update

### ISSUE THREAD

**Issue by Travia21** [CleverRaven/Cataclysm-DDA#50915]
### Describe the bug

Melee attack rolls multiplier doesn't update when dropping and wearing backpack. For me it's currently stuck at 0.79 regardless of whether or not I'm wearing a Leather Backpack. Torso encumbrance goes from 34 to 9, but multiplier doesn't change.

### Steps To Reproduce

1. Wear Leather Backpack with blanket and sneakers
2. Check encumbrance and melee attack roll modifier
2. Drop the Leather Backpack
3. Check encumbrance and melee attack roll modifier again

### Expected behavior

The melee attack roll modifier should go up or down based on current torso encumbrance.

### Screenshots

![image](https://user-images.githubusercontent.com/3219079/130171289-5fc95015-aa80-4633-b3e8-9fc3a3320b4b.png)
![image](https://user-images.githubusercontent.com/3219079/130171364-ccb68a00-8a8b-4943-a2a2-8b5d54f5d0f8.png)

### Versions and configuration

- OS: Windows 10
- Game Version: Experimental - 0.F - ea1f5e1
- Graphics version: Tiles
- Ingame language: English
- Mods loaded: DDA, Disable NPC Needs, Blaze Industries

### Additional context

[Travia.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/7018798/Travia.zip)

**Comment by actual-nh** [comment]
Ping: @KorGgenT?

**Comment by Travia21** [comment]
The bug might have something to do with the combat boots I'm wearing. They're the only piece of clothing that allows the melee attack roll stat to change now. I take them off and it goes to 1.00, even when I load up on slings and backpacks.

**Comment by Travia21** [comment]
Found the bug: src/character.cpp, line 9371 in the Character::melee_attack_roll_modifier() function.

I assume instead of returning the minimum of 0.2f and _balance_score()_ , it's supposed to be 0.2f and 1 - (torso encumbrance/100)

**Comment by VincentRoyceMorgan** [comment]
Yep, it's totally boots. I guess I'll either have to wait for a fix or get used to fighting without winter boots in the dead of winter. That's on top of the general movement cost I've noticed with my movement speed going from 115 to 145 with no change in gear after an update. Good times.

**Comment by Travia21** [comment]
As a workaround, you can edit character.cpp and build it yourself. Give me a minute and I'll figure out how to make the fix. I'm just not familiar enough with the codebase to tell you how at the moment...

**Comment by KorGgenT** [comment]
currently torso does not change melee attack roll, so yes it's a display bug. if you like, you can change the json in the bodyparts to a higher number on torso with lower numbers on other parts

**Comment by Travia21** [comment]
It's more than just a display bug. With 4 backpacks equipped and no shoes I was able to hit very consistently.

**Comment by RynKage** [comment]
It isn't just a display bug. The melee attack roll modifier seems to instead scale with foot encumbrance.

**Comment by toveri** [comment]
I have this same issue, I also noticed that the minimum melee attack roll modifier changed to 0.2 instead of the old 0.25. Was this an oversight?

[https://github.com/CleverRaven/Cataclysm-DDA/pull/50829/files#diff-2cd8dac3d0903ca1ee60a47c28e09bcb56290214bc8d1fe5cf4afaed108b9d92L328](https://github.com/CleverRaven/Cataclysm-DDA/pull/50829/files#diff-2cd8dac3d0903ca1ee60a47c28e09bcb56290214bc8d1fe5cf4afaed108b9d92L328)

**Comment by kurtisvetter** [comment]
As a temporary workaround, I modified the functions in character_modifier.cpp which determine the melee attack rolls and melee attack time. I think this goes against the spirit of #50829 by @KorGgenT and it seems like there are other changes which I don't understand, so I'll just leave it here.

```c++
float Character::melee_attack_roll_modifier() const
{
    return std::max( 0.2f, 1.0f - avg_encumb_of_limb_type( body_part_type::type::torso ) / 100.0f );
}
```
```c++
float Character::melee_thrown_move_modifier_torso() const
{
    if( avg_encumb_of_limb_type( body_part_type::type::torso ) == 0 ) {
        return MAX_MOVECOST_MODIFIER;
    } else {
        return std::min( MAX_MOVECOST_MODIFIER, 100.0f / ( 100.0f - avg_encumb_of_limb_type( body_part_type::type::torso ) ) );
    }
}
```
Changing these functions will make torso encumbrance work like it did before.

**Comment by Travia21** [comment]
Awesome, thanks. At the very least we can make the modification ourselves while we wait for an update on latest experimental.

**Comment by wapcaplet** [comment]
See also #52418

**Comment by Travia21** [comment]
Fixed by [#53070](https://github.com/CleverRaven/Cataclysm-DDA/pull/53070)
